src/templates/next/components/Navbar/Navbar.tsx

```diff
@@ -1,178 +1,285 @@
-              <div className="relative flex h-16 justify-between">
-                <div className="absolute inset-y-0 left-0 flex items-center sm:hidden">
-                  {/* Mobile menu button */}
-                  <Disclosure.Button className="relative inline-flex items-center justify-center rounded-md p-2 text-gray-400 hover:bg-gray-100 hover:text-gray-500 focus:outline-none focus:ring-2 focus:ring-inset focus:ring-indigo-500">
-                    <span className="absolute -inset-0.5" />
-                    <span className="sr-only">Open main menu</span>
-                    {open ? (
-                      <XMarkIcon className="block h-6 w-6" aria-hidden="true" />
-                    ) : (
-                      <Bars3Icon className="block h-6 w-6" aria-hidden="true" />
-                    )}
-                  </Disclosure.Button>
-                </div>
-                <div className="flex flex-1 items-center justify-center sm:items-stretch sm:justify-start">
-                  <div className="flex flex-shrink-0 items-center">
-                    <LinkComponent href="/">
-                      <img
-                        className="h-8 w-auto"
-                        src={logo.url}
-                        alt={logo.alt}
-                      />
+    <div className="flex flex-col w-full">
+      <div className="flex flex-row gap-5 px-14 xl:px-0 py-6 mx-auto max-h-24 w-full xl:max-w-screen-xl">
+        {/* Logo */}
+        <LinkComponent href="/">
+          <img
+            src={logoUrl}
+            alt={logoAlt}
+            className="h-full w-full max-w-48 object-cover object-center"
+          />
+        </LinkComponent>
+
+        {/* Spacer */}
+        <div className="flex-1" />
+
+        {/* Search */}
+        {search && search.type === "localSearch" && !isHamburgerOpen && (
+          <form
+            action={search.searchUrl}
+            method="get"
+            className={`flex flex-row gap-2 h-10 ${
+              isSearchOpen ? "hidden lg:flex" : ""
+            }`}
+          >
+            {isSearchOpen && (
+              <input
+                type="search"
+                name="q"
+                placeholder="Search this site"
+                className="w-96 px-4 py-2 border border-divider-medium focus:border-site-primary focus:ring-site-primary focus:outline-none hidden lg:block"
+              />
+            )}
+            <button
+              onClick={(e) => {
+                if (!isSearchOpen) {
+                  e.preventDefault()
+                  setIsSearchOpen(true)
+                }
+              }}
+              type="submit"
+              aria-label="Search this site"
+            >
+              <BiSearch className="text-2xl mt-0.5" />
+            </button>
+          </form>
+        )}
+
+        {/* Hamburger menu for small screens */}
+        <button className="block xl:hidden h-10">
+          {isHamburgerOpen ? (
+            <button
```

Comment:
we seem to be nesting a `button` within another `button`?

i understand that the above button seems to be hidden on small screens but this nevertheless bad behaviour. could we change it?

Reply:
Adjusted in 33a5297969cb617a3746f06d60fcfb9ee361f6c1 to use `<div>` instead on the parent, missed this out when changing.